lxd/storage/drivers/driver_ceph_volumes.go

```diff
@@ -855,7 +855,18 @@ func (d *ceph) commonVolumeRules() map[string]func(value string) error {
 
 // ValidateVolume validates the supplied volume config.
 func (d *ceph) ValidateVolume(vol Volume, removeUnknownKeys bool) error {
-	return d.validateVolume(vol, d.commonVolumeRules(), removeUnknownKeys)
+	commonRules := d.commonVolumeRules()
+
+	// Disallow block.* settings for regular custom block volumes. These settings only make sense
+	// when using custom filesystem volumes with block mode enabled. LXD will create the filesystem
+	// for these volumes, and use the mount options. When attaching a regular block volume to a VM,
+	// these are not mounted by LXD and therefore don't need these config keys.
+	if vol.IsVMBlock() || vol.volType == VolumeTypeCustom && vol.contentType == ContentTypeBlock {
```

Comment:
Im not following why we need this, ceph filesystem volumes are always block backed (like lvm) so they dont have "block mode". This looks wrong to me.

Reply:
Please can you explain the problem, as maybe this is needed, but we will need to change the comment at least to not reference "block mode" which isnt a thing for ceph and lvm.

Reply:
When you create a custom storage volume of type block using `lxc storage volume create {ceph/lvm} vol --type block block.filesystem={fs}` all of the drivers except LVM and Ceph are currently rejecting the request. 

This makes sense since LXD should not manage a FS on top of a `--type block` volume. In fact LXD doesn't for ZFS (block mode), LVM and Ceph. But when it comes to restoring volumes Ceph is missing a check for `vol.contentType == ContentTypeFS` to not trigger the restore on `--type block` volumes. The other drivers handle it accordingly.

Overall this wouldn't be a problem if the drivers wouldn't append the `block.filesystem` config key to `--type block` volumes. Regardless of the volumes content type Ceph driver is only checking for `block.filesystem` to decide if the FS UUID needs regeneration. That is fixed in this PR. Additionally the `block.*` settings aren't anymore appended to `--type block` volumes and creating such a volume with those being set will now fail.

I'll get rid of ZFS block mode in the comments.

Reply:
Excellent thanks! Makes sense now.